test/usecases/global/test_non_threadsafe.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+
+from neuron import h, gui
+from neuron.units import ms
+
+
+def test_non_threadsafe():
+    nseg = 1
+
+    s = h.Section()
+    s.insert("non_threadsafe")
+    s.nseg = nseg
+
+    h.finitialize()
+
+    instance = s(0.5).non_threadsafe
+
+    # Check INITIAL values.
+    assert instance.get_parameter() == 41.0
+    assert instance.get_gbl() == 42.0
+    assert instance.get_top_local() == 43.0
+
+    # Check reassigning a value. Top LOCAL variables
+    # are not exposed to HOC/Python.
```

Comment:
Could that be tested `with pytest.raises`?

Reply:
Yes, good point. However, nicely systematically testing visibility can be a separate task.

Reply:
https://github.com/BlueBrain/nmodl/issues/1421

(however, not with `pytest`, because I absolutely want to be able to run these without pytest.)